libraries/equativUtils/equativUtils.js

```diff
@@ -28,3 +54,102 @@ export function getBidFloor (bid, currency, mediaType) {
 
   return floors.length ? Math.min(...floors) : DEFAULT_FLOOR;
 }
+
+/**
+ * Returns a floor price provided by the Price Floors module or the floor price set in the publisher parameters
+ * @param {*} bid
+ * @param {string} mediaType A media type
+ * @param {number} width A width of the ad
+ * @param {number} height A height of the ad
+ * @param {string} currency A floor price currency
+ * @returns {number} Floor price
+ */
+function getFloor(bid, mediaType, width, height, currency) {
+  return bid.getFloor?.({ currency, mediaType, size: [width, height] })
+    .floor || bid.params.bidfloor || -1;
+}
+
+/**
+ * Generates a 14-char string id
+ * @returns {string}
+ */
+function makeId() {
+  const length = 14;
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  let counter = 0;
+  let str = '';
+
+  while (counter++ < length) {
+    str += characters.charAt(Math.floor(Math.random() * characters.length));
+  }
+
+  return str;
+}
+
+
+/**
+ * Prepares impressions for the request
+ *
+ * @param {*} imps An imps array
+ * @param {*} bid A bid
+ * @param {string} currency A currency
+ * @param {*} impIdMap An impIdMap
+ * @param {string} adapter A type of adapter (may be 'stx' or 'eqtv')
+ * @return {*}
+ */
+export function prepareSplitImps(imps, bid, currency, impIdMap, adapter) {
+  const splitImps = [];
+
+  imps.forEach(item => {
+    const floorMap = {};
+
+    const updateFloorMap = (type, name, width = 0, height = 0) => {
+      const floor = getFloor(bid, type, width, height, currency);
+
+      if (!floorMap[floor]) {
+        floorMap[floor] = {
+          ...item,
+          bidfloor: floor
+        };
+      }
+
+      if (!floorMap[floor][name]) {
+        floorMap[floor][name] = type === 'banner' ? { format: [] } : item[type];
+      }
+
+      if (type === 'banner') {
+        floorMap[floor][name].format.push({ w: width, h: height });
+      }
+    };
+
+    if (item.banner?.format?.length) {
+      item.banner.format.forEach(format => updateFloorMap('banner', 'bannerTemp', format?.w, format?.h));
+    }
+
+    updateFloorMap('native', 'nativeTemp');
+    updateFloorMap('video', 'videoTemp', item.video?.w, item.video?.h);
+
+    Object.values(floorMap).forEach(obj => {
+      [
+        ['banner', 'bannerTemp'],
+        ['native', 'nativeTemp'],
+        ['video', 'videoTemp']
+      ].forEach(([name, tempName]) => obj = cleanObject(obj, name, tempName));
+
+      if (obj.banner || obj.video || obj.native) {
+        const id = makeId();
+        impIdMap[id] = obj.id;
+        obj.id = id;
+
+        if (obj.banner && adapter === 'stx') {
```

Comment:
This statement is not covered by u.t. Maybe it's a good idea to create a spec file in test/spec/libraries/equativUtils/ and add a test just for the method `prepareSplitImps`? It's not a part of this ticket but each method here should be tested separately and only in adapters' tests.